common/flatpak-dir.c

```diff
@@ -4726,50 +4753,72 @@ flatpak_dir_pull (FlatpakDir                           *self,
 
           context = flatpak_main_context_new_default ();
 
-          ostree_repo_find_remotes_async (self->repo, (const OstreeCollectionRef * const *) collection_refs_to_fetch,
+          ostree_repo_find_remotes_async (repo, (const OstreeCollectionRef * const *) collection_refs_to_fetch,
                                           find_options,
                                           NULL /* default finders */, progress, cancellable,
                                           async_result_cb, &find_result);
 
           while (find_result == NULL)
             g_main_context_iteration (context, TRUE);
 
-          allocated_results = ostree_repo_find_remotes_finish (self->repo, find_result, error);
+          allocated_results = ostree_repo_find_remotes_finish (repo, find_result, error);
 
           results = (const OstreeRepoFinderResult * const *) allocated_results;
           if (results == NULL)
-            return FALSE;
+            goto out;
 
-          for (i = 0, rev = NULL; results[i] != NULL && rev == NULL; i++)
-            rev = g_strdup (g_hash_table_lookup (results[i]->ref_to_checksum, &collection_ref));
+          /* Go ahead and pull commit metadata now, so that if any of the
+           * signatures are bad we can fall back to another rev from another
+           * result. This also means the flatpak_dir_setup_extra_data() call
+           * below won't need to fetch the metadata. */
```

Comment:
Maybe link to one of the issues which describes the scenario you’re protecting against, since you’ve got some good writeups of the whole issue here on github.

Reply:
Added a link